bin/cylc-wrap-func

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python2
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+INTERNAL USE (asynchronous xtrigger function execution).
+
+USAGE: cylc wrap-func <func-name> <'json-func-args'> <'json-func-kwargs'>
+
+Run "func-name(*func-args, **func-kwargs)" in the command process pool.
+
+The function must be defined in a module of the same name (func-name).
+
+The function's stdout is redirected to stderr, and visible in suite log in
+debug mode. Its return value is printed to stdout as a JSON string.
+"""
+
+# Function return value to stdout: for compat with the command process pool.
+
+import importlib
+import json
+import sys
+from cylc.xtrigger_mgr import get_func
+
+
+def run_func(func_name, func_args, func_kwargs):
+    func = get_func(func_name)
+    # Redirect stdout to stderr.
+    orig_stdout = sys.stdout
+    sys.stdout = sys.stderr
+    res = func(*func_args, **func_kwargs)
+    # Restore stdout.
+    sys.stdout = orig_stdout
+    # Write function return value as JSON to stdout.
+    sys.stdout.write(json.dumps(res))
+
+
+if __name__ == "__main__":
+    if sys.argv[1] in ['help', '--help']:
+        print __doc__
+        sys.exit(0)
+    func_name = sys.argv[1]
+    func_args = json.loads(sys.argv[2])
+    func_kwargs = json.loads(sys.argv[3])
+    run_func(func_name, func_args, func_kwargs)
```

Comment:
It may be good to move the *json-load => call function => json-dump* logic to a location under `lib/` (in the future?). (See also protocols such as JSON-RPC.)

Reply:
Could do (pref "for the future" as I haven't time to look at JSON-RPC right now)

Reply:
Fair enough.

Reply:
I've taken your advice on move into `lib/cylc`.